### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by kalutheo** [MattCheely/generator-elm-parcel#4]
This issue will help us discuss about a clear Roadmap for the project. 
I am suggesting adding a Roadmap section on the Readme.md of the project. 
@MattCheely Maybe we can list Top features we would like to add here ?

**Comment by MattCheely** [comment]
I think a roadmap on the Reame.md, or maybe a link to a roadmap is a good idea.

We could also try creating a 1.0 milestone and filing issues under that. I haven't worked much with the issue management tools in Github, so I don't know how easy it is to manage things that way.

These are the short term goals I am thinking about:
- Fix #3 
- Rename to elm-app-gen
- Add support from multiple Elm Program types like in your generator (this is the first big feature addition I'd like to see, feel free to PR any time)

I have just been working out of the `standalone` branch with the plan to merge to master once the CLI version is ready to publish to npm. That way the Github repo still lines up with what's in npm. Does that work for you?

After that, these are the additional generators I am interested in:
- Create a generator for localstorage interop via ports
- Create a generator for websocket interop via ports

Previously I think you mentioned these, both of which I think would be nice to have:
- Generator for using elm-css
- Generator for elm-ui

Any other features you'd like to see?

**Comment by MattCheely** [comment]
I had another idea for a feature that I think might be a nice long-term addition: adding integration with tool like https://github.com/malaire/elm-dependencies-analyzer to make upgrading (and maybe finding and installing?) packages easier.

**Comment by kalutheo** [comment]
I think the list of features is fine. i prefer to have few ones but solid ones :) 

i will handle this one  : 
> Add support from multiple Elm Program types like in your generator (this is the first big feature addition I'd like to see, feel free to PR any time)

Regarding the Dependencies Analyzer i see that as a nice to have but not mendatory. We should focus on the core features in my opinion

**Comment by MattCheely** [comment]
I agree about dependency management being non-essential, and not a focus right now. I just wanted to get it down somewhere so I wouldn't forget the idea.
